server/pulp/server/db/model/__init__.py

```diff
@@ -631,7 +631,7 @@ def unit_key_str(self):
         """
         The unit key represented as a string ordered by unit key fields alphabetically
         """
-        return str(sorted([getattr(self, key) for key in self.unit_key_fields]))
+        return str(sorted(self.unit_key.items()))
```

Comment:
Good point.  The `__str__()` representation of a list of tuples is not guaranteed.
I will pursue your suggestion further before merging.